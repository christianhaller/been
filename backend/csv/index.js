var json2csv = require('json2csv'),
    Promise = require('promise'),
    fields = ['lat', 'lon', 'name', 'been'];

   /* upload = function (filename, csv, cb) {
        compress(csv).then(function(data){
            var params = {
                    Bucket: bucketName,
                    Key: 'csv/' + filename,
                    Body: data,
                    ACL: 'public-read',
                    ContentEncoding: 'gzip',
                    ContentType: 'text/csv'
                },

                upload = new AWS.S3.ManagedUpload({params: params});
            upload.send(function (err, data) {
                if (err) {
                    console.log(err);
                }
                else {
                    console.log(data);
                    cb(data.Location);
                }
            });
        });


    };*/

module.exports = function (data) {
    return new Promise(function (fulfill,reject) {
    var map = [];
        data.places.forEach(function (item) {
            item.been = item.flags.join(',');
            item.lon = item.lng;
            map.push(item);
        });

        json2csv({data: map, fields: fields}, function (err, csv) {
            if (err) {

                console.log('err');
                reject(err);

            }
            else {
                console.log('ok');
                fulfill(csv);
            }

        });
    });
};


