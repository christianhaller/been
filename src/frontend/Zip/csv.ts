import json2csv from "json2csv/dist/json2csv.umd";
import type { EnhancedPin } from "../../backend/interace";

export function csv(data: EnhancedPin[]) {
  const fields = ["lat", "lon", "country", "city", "been"];
  return json2csv.parse(
    data.map(({ flags, lng, lat, country, city }) => {
      return {
        been: flags.join(","),
        lon: lng,
        lat,
        country,
        city,
      };
    }),
    {
      delimiter: ";",
      quote: "",
      fields,
    }
  );
}
