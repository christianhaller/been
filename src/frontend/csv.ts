import json2csv from "json2csv/dist/json2csv.umd";

export function csv(data) {
  const fields = ["lat", "lon", "name", "country", "city", "been"];
  return json2csv.parse(
    data.map(({ flags, lng, lat, name, country, city }) => {
      return {
        been: flags.join(","),
        lon: lng,
        lat,
        name,
        country,
        city,
      };
    }),
    {
      fields,
    }
  );
}
