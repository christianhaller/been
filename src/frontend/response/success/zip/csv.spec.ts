import { csv } from "./csv";

test("csv", () => {
  const pins = [
    {
      lat: 50.444885,
      lng: 30.536667,
      flags: ["been"],
      city: "Kyiv(Kiev)",
      country: "Ukraine",
    },
    {
      lat: -34.45451,
      lng: -57.83755,
      flags: ["been"],
      city: "Colonia del Sacramento",
      country: "Uruguay",
    },
  ];
  const res = csv(pins);
  expect(res).toMatchSnapshot();
});
