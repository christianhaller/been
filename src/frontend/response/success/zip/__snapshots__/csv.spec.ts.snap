// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csv 1`] = `
"lat;lon;country;city;been
50.444885;30.536667;Ukraine;Kyiv(Kiev);been
-34.45451;-57.83755;Uruguay;Colonia del Sacramento;been"
`;
